framework/src/service/ServiceHooks.cpp

```diff
@@ -75,10 +74,8 @@ void ServiceHooks::RemovedService(const ServiceReference<ServiceListenerHook>& /
 
 void ServiceHooks::Open()
 {
-  {
-    auto l = this->Lock(); US_UNUSED(l);
-    listenerHookTracker.reset(new ServiceTracker<ServiceListenerHook>(GetBundleContext(), this));
-  }
+  auto l = this->Lock(); US_UNUSED(l);
+  listenerHookTracker.reset(new ServiceTracker<ServiceListenerHook>(GetBundleContext(), this));
```

Comment:
Yes, you're right. I forgot about the call back functions.

I was looking at this code with your suggestion in mind and a question popped into my head; does ```listenerHookTracker``` need to be ```reset()``` on each ```ServiceHooks:::Open()``` and ```ServiceHooks::Close()``` call? What if ```listenerHookTracker``` was only created once and allowed to be destroyed during the ```ServiceHooks``` object clean-up? 
Is there state within ```listenerHookTracker``` which requires re-creating the object each time ```ServiceHooks::Open()``` is called?